pkg/backend-storage/backend-storage.go

```diff
@@ -65,11 +77,63 @@ func CreateIfNeeded(vmi *corev1.VirtualMachineInstance, clusterConfig *virtconfi
 			VolumeMode:       &modeFile,
 		},
 	}
+	// Whatever owns the VMI (i.e. a VM) now owns the PVC
```

Comment:
When the VM obejct is deleted, do we need also to delete the PVC?

Reply:
Yes. I'm not sure how this mechanism works, I'll look into it.

Reply:
The PVC indeed gets deleted with the VM, I added a test for it.

Reply:
ok, what happen if we simpy create a VMI? This doesn't have a owner reference. What happen to the PVC in this case?

Reply:
In that case the PVC stays, since we don't know if/when that VMI will be started again...
The alternative is to prevent the creation of VMIs with persistent settings enabled.

Reply:
So while the VMI is shut down, if I start another VMI with the same name it will pick the existing PVC with TPM state, right? I am not deeply familiar with TPM, but does that mean that from the newly created VMI I can in theory get access to all the stored secrets?

Reply:
Potentially yes. Same thing if a VM if stopped and you create a VMI with its name.
However:
- If the libvirt UUID is different, the existing TPM will be wiped out and a blank one will be created
- If the original VMI took ownership of the TPM, the new VMI will need the administrator password to do anything on it
- If the secrets were sealed against a set of PCRs that are unique to the original VMI, the new one will not be able to see them
- Most importantly, this should be seen as a convenience feature rather than a security one. The whole premise of TPM is the hardware vault protecting its flash memory...

Reply:
I see. Thanks for clarification. So:

> If the libvirt UUID is different, the existing TPM will be wiped out and a blank one will be created

This basically means that there is no point in allowing persistent TPM settings for VMIs since UUID will be different every time, am I right? 

And BTW, just curious: I never checked that, but if I create a VM, then each time that I start/stop it the UUID will be the same, right?

Reply:
I thought a new UUID was regenerated each time. Interesting...

Reply:
Indeed, the libvirt docs says a random UUID is generated for each new domain (and every boot of a VM/VMI effectively creates a new domain).
However, I found this, though it is specific to VMs:
https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/watch/vm.go#L1395

I just tried to verify my claim that VMIs keep getting the same UUID and actually got different values across reboots! Not sure what changed or maybe my initial observation was bogus...
However, even for VMIs we do request a specific UUID in the domain XML, so we have control over it. Good to know in case we want to ensure it's always the same value or something.